cmd/cli/install.go

```diff
@@ -181,7 +187,10 @@ func newInstallCmd(config *helm.Configuration, out io.Writer) *cobra.Command {
 	f.BoolVar(&inst.enableFluentbit, "enable-fluentbit", defaultEnableFluentbit, "Enable Fluentbit sidecar deployment")
 	f.StringVar(&inst.meshName, "mesh-name", defaultMeshName, "name for the new control plane instance")
 	f.BoolVar(&inst.deployJaeger, "deploy-jaeger", defaultDeployJaeger, "Deploy Jaeger in the namespace of the OSM controller")
-	f.StringVar(&inst.envoyLogLevel, "envoy-log-level", defaultEnvoyLogLevel, "Envoy log level is used to specify the level of logs collected from envoy and needs to be one of these (trace, debug, info, warning, warn, error, critical, off)")
+	f.StringVar(&inst.envoyLogLevel, "envoy-log-level", defaultEnvoyLogLevel,
+		fmt.Sprintf("Envoy log level is the level of logs collected from envoy and needs to be one of: %s", strings.Join(possibleEnvoyLogLevels, ",")))
+	f.StringVar(&inst.controllerLogLevel, "controller-log-level", defaultControllerLogLevel,
+		fmt.Sprintf("Controller log level is the level of logs collected from envoy and needs to be one of these: %s", strings.Join(logger.AllowedLevels, ",")))
```

Comment:
@nojnhuh is it possible to use the `--set` option and examine the values set for correctness, as opposed to using a separate command line option for each chart parameter?

Reply:
When validation is defined in values.schema.json, then a value from `--set` will be validated against it. That's the easiest way for simple regex checks like this. If something more complex is needed that needs to be done in Go, then there's no established pattern to do that for values from `--set` currently. It's certainly possible.

Reply:
Okay, then why not use `--set` flag to override osm-controller log value instead of defining a cli flag? Its better to be consistent in how chart values are overridden using the cli. Some of the older commands (ex. `envoy-log-level`) were implemented when `--set` didn't exist.

Reply:
The main advantage I see with command line flags is the CLI will error if you make a typo in the flag name but a typo in the key when using `--set` doesn't (helm just silently accepts it as a new value that doesn't do anything). Command line flags also show up in `--help`. We could perhaps show all the chart values, defaults, and descriptions in `--help`, but I'd have to play with it to see if that would work.

Overall I agree that splitting the values between "real" flags and `--set` isn't ideal, but I can see use cases for both.

Reply:
I just think there needs to be some guidance around when `--set` is acceptable vs a separate cli flag. Mixing the two of them doesn't give a consistent user experience, especially because one could use the `--set` option and attempt to override a chart value that already has a CLI flag defined.

Reply:
@draychev can we instead use `--set OpenServiceMesh.controllerLogLevel=foo` to override the chart values instead of bloating the CLI? We have already started moving in that direction. Some of the legacy options can move to this model as well.